src/ruby/lib/grpc/errors.rb

```diff
@@ -47,7 +48,20 @@ def initialize(code, details = 'unknown cause', metadata = {})
     #
     # @return [Status] with the same code and details
     def to_status
-      Struct::Status.new(code, details, @metadata)
+      Struct::Status.new(code, details, metadata)
+    end
+
+    # Converts the exception to a deserialized Google::Rpc::Status proto.
+    # Returns nil if the `grpc-status-details-bin` trailer could not be
+    # converted to a {Google::Rpc::Status} due to the server not providing
+    # the necessary trailers.
+    #
+    # Raises an error if the server did provide the necessary trailers
+    # but they fail to deserialize into a {Google::Rpc::Status} protobuf.
+    #
+    # @return [Google::Rpc::Status] with the same code and details
+    def to_rpc_status
+      GoogleRpcStatusUtils.extract_google_rpc_status to_status
```

Comment:
Right now `GoogleRpcStatusUtils.extract_google_rpc_status to_status` will raise if the metadata doesn't contains a serialized protobuf. I think it makes sense for `GoogleRpcStatusUtils.extract_google_rpc_status to_status` to raise, but I don't think it makes sense for `BadStatus#to_rpc_status` to raise. Would it be acceptable for this method to rescue from an error and return `nil` instead?

Reply:
It's basically a server-side error if this fails to deserialize AFAICS, are we seeing this happen against real services?

In any case, I'd tend to want to avoid a change like that, just since it's breaking. I different utility that has those semantics might be OK, but first I wonder if that raise is happening with real services.

Reply:
Well, since `BadStatus#to_rpc_status` doesn't exist, it isn't a breaking change. I don't mean to change the behavior of `GoogleRpcStatusUtils.extract_google_rpc_status`, I just mean to have `BadStatus#to_rpc_status` handle the exception and return `nil`. See my updated commits for what I mean.

Reply:
I see, missed that. I think it's fine to have a different semantic for `to_rpc_status`